### PR TITLE
Change players structure to allow hash to allow players to rename the…

### PR DESCRIPTION
**Pull request by jenf**
…mselves (#1330)

### CODE REVIEW

**Comment by michaeljb** [comment]
Screenshot from 2020-11-15 11-06-18

Started a local 1846 game, the id is showing up alongside the player names during the Draft step, and clicking "Show Companies" doesn't work, leaving this error in the console:

```
Uncaught TypeError: a['$=='] is not a function
    _eqeq http://localhost:9292/assets/opal.js:11700
    $Array_$eq_eq$11 http://localhost:9292/assets/opal.js:11709
    $Player_$eq_eq$7 http://localhost:9292/assets/engine.js:36807
    $$render_info http://localhost:9292/assets/app.js:11662
    $$render_body http://localhost:9292/assets/app.js:11634
    $$render http://localhost:9292/assets/app.js:11615
    $$h http://localhost:9292/assets/deps.js:2693
    $$render_players http://localhost:9292/assets/app.js:3759
    $$render http://localhost:9292/assets/app.js:3746
    $$h http://localhost:9292/assets/deps.js:2693
    render_action http://localhost:9292/assets/app.js:13897
    $$render_action http://localhost:9292/assets/app.js:13900
    $$render_game http://localhost:9292/assets/app.js:13907
    page http://localhost:9292/assets/app.js:13730
    $$render http://localhost:9292/assets/app.js:13738
    $$h http://localhost:9292/assets/deps.js:2693
    $$render_game http://localhost:9292/assets/app.js:629
    page http://localhost:9292/assets/app.js:585
    $$render_content http://localhost:9292/assets/app.js:594
    $$render http://localhost:9292/assets/app.js:569
    $Component_update$excl$11 http://localhost:9292/assets/deps.js:2730
    update http://localhost:9292/assets/deps.js:2710
    $$update http://localhost:9292/assets/deps.js:2710
    $$store http://localhost:9292/assets/deps.js:2790
    $$7 http://localhost:9292/assets/app.js:3797
    invokeHandler http://localhost:9292/assets/deps.js:2141
    handleEvent http://localhost:9292/assets/deps.js:2155
    handler http://localhost:9292/assets/deps.js:2161
    updateEventListeners http://localhost:9292/assets/deps.js:2205
    createElm http://localhost:9292/assets/deps.js:1719
    createElm http://localhost:9292/assets/deps.js:1726
    createElm http://localhost:9292/assets/deps.js:1726
    createElm http://localhost:9292/assets/deps.js:1726
    updateChildren http://localhost:9292/assets/deps.js:1864
    patchVnode http://localhost:9292/assets/deps.js:1909
    updateChildren http://localhost:9292/assets/deps.js:1836
    patchVnode http://localhost:9292/assets/deps.js:1909
    patch http://localhost:9292/assets/deps.js:1940
    yieldX http://localhost:9292/assets/opal.js:1475
    $$call http://localhost:9292/assets/opal.js:17636
    $Component_update$excl$11 http://localhost:9292/assets/deps.js:2731
    update http://localhost:9292/assets/deps.js:2710
    $$update http://localhost:9292/assets/deps.js:2710
    $$store http://localhost:9292/assets/deps.js:2790
    $$19 http://localhost:9292/assets/app.js:218
    yieldX http://localhost:9292/assets/opal.js:1475
    $$call http://localhost:9292/assets/opal.js:17636
    $$7 http://localhost:9292/assets/app.js:1050
    send http://localhost:9292/assets/app.js:1126
    promise callback*$$send http://localhost:9292/assets/app.js:1124
    $$get http://localhost:9292/assets/app.js:1011
    send http://localhost:9292/assets/opal.js:1671
    $$safe_get http://localhost:9292/assets/app.js:1039
    send http://localhost:9292/assets/opal.js:1671
    $$enter_game http://localhost:9292/assets/app.js:207
    $$render_game http://localhost:9292/assets/app.js:619
    page http://localhost:9292/assets/app.js:585
    $$render_content http://localhost:9292/assets/app.js:594
    $$render http://localhost:9292/assets/app.js:569
    $Component_update$excl$11 http://localhost:9292/assets/deps.js:2730
    update http://localhost:9292/assets/deps.js:2710
    $$update http://localhost:9292/assets/deps.js:2710
    $$attach http://localhost:9292/assets/deps.js:2620
    <anonymous> http://localhost:9292/game/16033?debug=1:24
```

**Review by michaeljb — DISMISSED** [review]
Screenshot from 2020-11-15 11-06-18

Started a local 1846 game, the id is showing up alongside the player names during the Draft step, and clicking "Show Companies" doesn't work, leaving this error in the console - https://gist.github.com/michaeljb/8ddb6b4fc50b3441fd88a87ccdf82a69

File: routes/game.rb [hunk 1]
```diff
@@ -80,9 +80,11 @@ class Api
 
                 game.save
               else
+                players = users.map { |u| [u.id, u.name] }.to_h
                 engine = Engine::GAMES_BY_TITLE[game.title].new(
-                  users.map(&:name),
+                  players,
                   id: game.id,
+                  settings: game.settings,
```
Comment (michaeljb) [inline routes/game.rb:87]: This line is still causing an error for me when taking an action

Screenshot from 2020-11-15 11-47-06

Comment (jenf) [reply routes/game.rb]: Thanks for the catch

**Review by michaeljb — APPROVED** [review]

**Pull request merged**